components/menu/index.css

```diff
@@ -10,170 +10,623 @@ OF ANY KIND, either express or implied. See the License for the specific languag
-      }
+    .spectrum-Menu-itemDescription {
+      line-height: var(--mod-menu-item-description-line-height-cjk, var(--spectrum-menu-item-description-line-height-cjk));
+    }
+
+    .spectrum-Menu-sectionHeading {
+      line-height: var(--mod-menu-section-header-line-height-cjk, var(--spectrum-menu-section-header-line-height-cjk));
     }
   }
+
   .spectrum-Menu-divider {
     overflow: visible;
-
     inline-size: auto;
-    margin-block: calc(var(--spectrum-listitem-texticon-divider-padding) / 2);
-    margin-inline: var(--spectrum-listitem-texticon-padding-y);
+    margin-block: var(--mod-menu-item-bottom-edge-to-text, var(--spectrum-menu-item-bottom-edge-to-text));
   }
 }
 
+.spectrum-Menu-itemIcon {
+  fill: var(--highcontrast-menu-item-label-icon-color-default, var(--mod-menu-item-label-icon-color-default, var(--spectrum-menu-item-label-icon-color-default)));
+}
+
 .spectrum-Menu-checkmark {
-  transform: scale(1);
+  align-self: center;
+  display: var(--mod-menu-checkmark-display, var(--spectrum-menu-checkmark-display));
+  fill: var(--highcontrast-menu-checkmark-icon-color-default, var(--mod-menu-checkmark-icon-color-default, var(--spectrum-menu-checkmark-icon-color-default)));
   opacity: 1;
 }
 
-.spectrum-Menu-item {
+.spectrum-Menu-item,
+.spectrum-Menu li:not(.spectrum-Menu-item, .spectrum-Menu-divider) {
   cursor: pointer;
   position: relative;
-  display: flex;
   align-items: center;
 
   box-sizing: border-box;
 
-  padding: var(--spectrum-listitem-texticon-padding-y)
-    var(--spectrum-listitem-texticon-padding-right)
-    var(--spectrum-listitem-texticon-padding-y)
-    var(--spectrum-listitem-texticon-padding-left);
+  background-color: var(--mod-menu-item-background-color-default, var(--spectrum-menu-item-background-color-default));
+  
+  line-height: var(--mod-menu-item-label-line-height, var(--spectrum-menu-item-label-line-height));
 
+  min-block-size: var(--mod-menu-item-min-height, var(--spectrum-menu-item-min-height));
+  padding-block-start: var(--mod-menu-item-top-edge-to-text, var(--spectrum-menu-item-top-edge-to-text));
+  padding-block-end: var(--mod-menu-item-bottom-edge-to-text, var(--spectrum-menu-item-bottom-edge-to-text));
+  padding-inline: var(--mod-menu-item-label-inline-edge-to-content, var(--spectrum-menu-item-label-inline-edge-to-content));
+  
   margin: 0;
+  text-decoration: none;
+}
 
-  border-inline-start: var(--spectrum-listitem-texticon-focus-indicator-size) solid
-    transparent;
+.spectrum-Menu-item,
+.spectrum-Menu li:not(.spectrum-Menu-item, .spectrum-Menu-divider) {
+  display: grid;
+  grid-template-areas:
+    ".            chevronAreaCollapsible .             iconArea sectionHeadingArea  .         .           .                 "
+    "selectedArea chevronAreaCollapsible checkmarkArea iconArea labelArea        valueArea actionsArea chevronAreaDrillIn"
+    ".            .                      .             .        descriptionArea  .         .           .                 "
```

Comment:
```suggestion
    ".            .                      .             iconArea       descriptionArea  .         .           .                 "
```
This would allow for layouts like (ignoring the misplaced check mark):
<img width="305" alt="image" src="https://user-images.githubusercontent.com/1156657/220625130-7ab7feda-487d-4efc-bfb0-5a5ff8f0aeb3.png">